ci/test-clang.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -eu
+
+trap 'rm -fr "$toolchain"' EXIT
+
+# Set up workspace
+rootdir=${GITHUB_WORKSPACE:-$(dirname "$(dirname "$(realpath "$0")")")}
+toolchain=$rootdir/toolchain
+
+# Pull toolchain out of container
+echo "[+] Downloading toolchain from container"
+docker create --name llvm-project ghcr.io/clangbuiltlinux/llvm-project:stage2
+mkdir -p "$toolchain"/{bin,include,lib}
+docker cp llvm-project:/usr/local/bin/. "$toolchain"/bin
+docker cp llvm-project:/usr/local/include/. "$toolchain"/include
+docker cp llvm-project:/usr/local/lib/. "$toolchain"/lib
+echo "[+] Cleaning up container"
+docker rm llvm-project
+
+# Test toolchain in Docker images so that the environment is consistent,
+# regardless of runners.
+docker_images=(
+    docker.io/fedora:latest
+)
+for docker_image in "${docker_images[@]}"; do
+    echo "[+] Updating '$docker_image'"
+    docker pull "$docker_image"
+    echo "[+] Testing clang in '$docker_image' container"
+    docker run --rm -v "$rootdir":/repo:ro "$docker_image" bash /repo/ci/test-clang-docker.sh
```

Comment:
so this will copy `test-clang-docker.sh` into `docker.io/fedora:latest`, then run it? We don't need an explict `docker cp` to `cp` `test-clang-docker.sh` into `docker.io/fedora:latest`.

Reply:
`-v` is short for `--volume`, which means that we mount the repository into the container at `/repo` and it is read only to make sure we don't do any modifications to it. I should probably use the long flag to make that clearer.

Reply:
please do, `-v` makes me think `--verbose`, which it's not for `docker run`.

Reply:
Will do.

Reply:
Done: 22c734e153fa0f87dba1ff5a6de2725cb21fad56